pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/customizations.yaml

```diff
@@ -0,0 +1,106 @@
+              desiredNumberScheduled = desiredNumberScheduled + statusItems[i].status.desiredNumberScheduled
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.numberReady ~= nil then
+              numberReady = numberReady + statusItems[i].status.numberReady
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.updatedNumberScheduled ~= nil then
+              updatedNumberScheduled = updatedNumberScheduled + statusItems[i].status.updatedNumberScheduled
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.numberAvailable ~= nil then
+              numberAvailable = numberAvailable + statusItems[i].status.numberAvailable
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.numberUnavailable ~= nil then
+              numberUnavailable = numberUnavailable + statusItems[i].status.numberUnavailable
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.observedGeneration ~= nil then
+              generation = statusItems[i].status.observedGeneration
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.daemonSetHash ~= nil then
+              daemonSetHash = statusItems[i].status.daemonSetHash
+            end
+          end
+          desiredObj.status.observedGeneration = generation
+          desiredObj.status.currentNumberScheduled = currentNumberScheduled
+          desiredObj.status.numberMisscheduled = numberMisscheduled
+          desiredObj.status.desiredNumberScheduled = desiredNumberScheduled
+          desiredObj.status.numberReady = numberReady
+          desiredObj.status.updatedNumberScheduled = updatedNumberScheduled
+          desiredObj.status.numberAvailable = numberAvailable
+          desiredObj.status.numberUnavailable = numberUnavailable
+          desiredObj.status.daemonSetHash = daemonSetHash
+          return desiredObj
+        end
+    statusReflection:
+      luaScript: >
+        function ReflectStatus (observedObj)
+          status = {}
+          if observedObj == nil or observedObj.status == nil then 
+            return status
+          end
+          status.observedGeneration = observedObj.status.observedGeneration
+          status.currentNumberScheduled = observedObj.status.currentNumberScheduled
+          status.numberMisscheduled = observedObj.status.numberMisscheduled
+          status.desiredNumberScheduled = observedObj.status.desiredNumberScheduled
+          status.numberReady = observedObj.status.numberReady
+          status.updatedNumberScheduled = observedObj.status.updatedNumberScheduled
+          status.numberAvailable = observedObj.status.numberAvailable
+          status.numberUnavailable = observedObj.status.numberUnavailable
+          status.daemonSetHash = observedObj.status.daemonSetHash
+          return status
+        end
+    healthInterpretation:

```

Comment:
The apps.kruise.io daemonset references the code in `pkg/resourceinterpreter/default/native/healthy.go` for the health part of the native daemonset, and furthermore ignores the  ` Status.ObservedGeneration == daemonSet.Generation` must be equal.

Reply:
> The apps.kruise.io daemonset references the code in `pkg/resourceinterpreter/default/native/healthy.go` for the health part of the native daemonset, and furthermore ignores the ` Status.ObservedGeneration == daemonSet.Generation` must be equal.

`InterpreterHealth` will be refined soon, and test report to be submitted.